infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt

```diff
@@ -298,19 +299,20 @@ class SplitService(
         target: SplitTarget,
         context: ValidationContext,
     ): Pair<List<AddressPoint>?, List<AddressPoint>?> {
-        val sourceMRange = getMRange(sourceTrack.getAlignmentVersionOrThrow(), target.segmentIndices)
```

Comment:
Varsinainen fix täällä. Tässä siis otettiin tasametripisteistä ne jotka osuu splitin m-arvovälille ja oletettiin että geokoodaamalla splitin target-raide, saadaan samat tasa-m-pisteet. Johtuen siitä että ne lasketaan liukuluvuilla kahta eri reittiä + siitä miten nuo luvut pyöristyy, tämä menee arpapeliksi jos reuna on ihan tasametripisteen vieressä.

Uudella logiikalla tuotetaan ensin osoiteväli geokoodaamalla source-raiteelta alku- ja loppukohta samoin kuin nuo geokoodataan myöhemmin target-raiteelta ja otetaan noiden osoitteiden väliin jäävät tasametripisteet. Koska laskentatapa on näin identtinen, pitäisi tuloksenkin olla.